habitat-baselines/habitat_baselines/rl/ppo/trainer_agent.py

```diff
@@ -0,0 +1,333 @@
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional
+
+import gym.spaces as spaces
+import numpy as np
+import torch
+from torch.optim.lr_scheduler import LambdaLR
+
+from habitat import logger
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.env_spec import EnvironmentSpec
+from habitat_baselines.common.rollout_storage import (  # noqa: F401.
+    RolloutStorage,
+)
+from habitat_baselines.rl.ddppo.policy import (  # noqa: F401.
+    PointNavResNetNet,
+    PointNavResNetPolicy,
+)
+from habitat_baselines.rl.hrl.hierarchical_policy import (  # noqa: F401.
+    HierarchicalPolicy,
+)
+from habitat_baselines.rl.ppo import PPO  # noqa: F401.
+from habitat_baselines.rl.ppo.policy import NetPolicy
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+@baseline_registry.register_agent
+class TrainerAgent:
+    """
+    A `TrainerAgent` consists of:
+    - Policy: How actions are selected from observations.
+    - Data Storage: How data collected from the environment is stored.
+    - Updater: How the Policy is updated.
+    """
+
+    def __init__(
+        self,
+        config: "DictConfig",
+        env_spec: EnvironmentSpec,
+        is_distrib: bool,
+        device,
+        resume_state: Optional[Dict[str, Any]],
+        num_envs: int,
+        percent_done_fn: Callable[[], float],
+        lr_schedule_fn: Optional[Callable[[float], float]] = None,
+    ):
+        """
+        :param percent_done_fn: Function that will return the percent of the
+            way through training.
+        :param lr_schedule_fn: For a learning rate schedule. ONLY used if
+            specified in the config. Takes as input the current progress in
+            training and returns the learning rate multiplier. The default behavior
+            is to use `linear_lr_schedule`.
+        """
+
+        self._env_spec = env_spec
+        self._config = config
+        self._num_envs = num_envs
+        self._device = device
+        self._ppo_cfg = self._config.habitat_baselines.rl.ppo
+        self._is_distributed = is_distrib
+        self._is_static_encoder = (
+            not config.habitat_baselines.rl.ddppo.train_encoder
+        )
+        self._actor_critic = self._create_policy()
+        self._actor_critic.to(self._device)
+        self._policy_action_space = self._actor_critic.get_policy_action_space(
+            env_spec.action_space
+        )
+        self._updater = self._create_updater(self._actor_critic)
+        if resume_state is not None:
+            self._updater.load_state_dict(resume_state["state_dict"])
+            self._updater.optimizer.load_state_dict(
+                resume_state["optim_state"]
+            )
+
+        logger.info(
+            "agent number of parameters: {}".format(
```

Comment:
```suggestion
            "Agent number of parameters: {}".format(
```
nit, is there any other information we want to give the user here ? Like the observation/action space ?